File: 1DAM/1DAM-Programacion/ProgramacionRepaso/AlumnosDAM/src/main/java/com/drodriguez/es/alumnosdam/repositories/CrudRepository.java
package com.drodriguez.es.alumnosdam.repositories;

import javafx.collections.ObservableList;

import java.sql.SQLException;
import java.util.Optional;

public interface CrudRepository<T, ID> {
    ObservableList<T> findAll() throws SQLException;


    Optional<T> findById(ID id) throws SQLException;


    Optional<T> save(T entity) throws SQLException;


    Optional<T> update(ID id,T entity) throws SQLException;


    void delete(T entity) throws SQLException;
}

File: 2DAM/AccesoADatos/Practicas/AD-03_TennisLab/AD-P03-TennisLab/TennisLab/src/main/kotlin/repositories/mongo/CrudRepository.kt
package repositories.mongo

import kotlinx.coroutines.flow.Flow

interface CrudRepository<T,ID> {
     fun findAll(): Flow<T>
    suspend fun findById(id: ID): T?
    suspend fun save(entity: T)
    suspend fun delete(_id: String)
    suspend fun update(entity: T)
}